packages/checkout-extension/src/ExtensionProvider.tsx

```diff
@@ -0,0 +1,28 @@
+import { CheckoutService } from '@bigcommerce/checkout-sdk';
+import React, { ReactNode, useState } from 'react';
+
+import { ExtensionService } from './ExtensionService';
+import { ExtensionContext } from './ExtensionContext';
+
+export interface ExtensionProviderProps {
+    checkoutService: CheckoutService;
+    children: ReactNode;
+}
+
+export const ExtensionProvider = ({ checkoutService, children }: ExtensionProviderProps) => {
+    const [isExtensionEnabled, setIsExtensionEnabled] = useState(false);
```

Comment:
Can we try using a `useReducer` for this?

Reply:
Will do.